tools/src/main/scala/scala/scalanative/build/TargetTriple.scala

```diff
@@ -0,0 +1,453 @@
+// ported from LLVM 887d6ab dated 2023-04-16
+
+//===--- Triple.cpp - Target triple helper class --------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+package scala.scalanative
+package build
+
+import java.nio.ByteOrder
+
+private[scalanative] final case class TargetTriple(
+    arch: String,
+    vendor: String,
+    os: String,
+    env: String
+) {
+  override def toString = s"$arch-$vendor-$os-$env"
+}
+
+private[scalanative] object TargetTriple {
+
+  def parse(triple: String): TargetTriple = {
+    val components = triple.split("-", 4).toList
+    val unknown = "unknown"
+    TargetTriple(
+      components.lift(0).map(Arch.parse).getOrElse(unknown),
+      components.lift(1).map(Vendor.parse).getOrElse(unknown),
+      components.lift(2).map(OS.parse).getOrElse(unknown),
+      components.lift(3).map(Env.parse).getOrElse(unknown)
+    )
+  }
```

Comment:
This parser is a bit naive e.g. it will not work correctly for e.g. `x86_64-linux-gnu` where the vendor is omitted. However it seems to work fine for the discovered target triple emitted by LLVM. If the user configures it manually they should be careful to specify all the parts e.g. `x86_64-unknown-linux-gnu`.

In the future we can do something more complex if necessary.